extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -18,7 +18,11 @@ export class WorkspaceService implements IWorkspaceService {
 	private _onDidWorkspaceProjectsChange: vscode.EventEmitter<void> = new vscode.EventEmitter<void>();
 	readonly onDidWorkspaceProjectsChange: vscode.Event<void> = this._onDidWorkspaceProjectsChange?.event;
 
-	constructor() { }
+	private openedProjects: vscode.Uri[] = [];
+
+	constructor() {
+		this.getProjectsInWorkspace(undefined, true);
```

Comment:
You may want to consider having re-entry protection for this stuff, currently you could potentially have issues if this is called multiple times in a row.